makepdf.sh

```diff
@@ -110,30 +146,143 @@ is_pandoc_version_new()
+                $(pandoc --pdf-engine=xelatex \
+                         --lua-filter $OLDPWD/pandoc/filter-web.lua \
+                         --metadata-file $OLDPWD/pandoc/config-web.yaml \
+                         -o $out_rel_path_file $input_basename".md")
+        
+        else
+                echo "ERROR: can only process one of these styles: opensolaris or web"
+                exit 1
+        fi
+}
+
+
 do_conversion()
 {
-        input=$1
-        output=$2
-        format=$3
-        style=$4
+        input_basename="$1"
+        format="$2"
+        style="$3"
 
+        
         if [ "$format" == "epub" ]; then
-                $(pandoc -t epub --toc -f markdown+grid_tables+table_captions -o $output $input  --pdf-engine=xelatex)
-        elif [ "$format" == "pdf" ]; then
+                do_conversion_epub "$input_basename" "$format"
+
+                
+        elif [ "$format" == "pdf" ]; then               
                 if [ "$(is_pandoc_version_new)" == "false" ]; then
-                        $(pandoc --toc -f markdown+grid_tables+table_captions -o $output $input  --pdf-engine=xelatex)
+                        # This call for backward compatibility 
+                        do_conversion_with_older_pandoc "$input_basename" "$format"
+                        
                 else
-                        if [ "$style" == "opensolaris" ]; then
-                            $(pandoc --pdf-engine=xelatex --lua-filter $OLDPWD/pandoc/filter-sb.lua --metadata-file $OLDPWD/pandoc/config-sb.yaml -o $output $input)
-                        else
-                            $(pandoc --pdf-engine=xelatex --lua-filter $OLDPWD/pandoc/filter-web.lua --metadata-file $OLDPWD/pandoc/config-web.yaml -o $output $input)
-                        fi
+                        do_conversion_pdf "$input_basename" "$format" "$style"
                 fi
         fi
+}
+
 
+# This function sets is_format_valid_result = True || False
+# True: users select a valid format
+# False: users select an invalid format
+is_format_valid()
+{
+        input_format=$1
+        is_format_valid_result="False"
+        
+        for  i in $SUPPORTED_FORMATS  ; do
+                if [ "${i}" == "$input_format" ]; then
+                        is_format_valid_result="True"
+                fi
+        done
 }
 
 
+make_out_dir()
+{
+        the_output_format=$1
+        path_to_input=$2
+        
+        is_format_valid $the_output_format
+        if [ "$is_format_valid_result" == "False" ]; then
+                echo >&2 "ERROR: -t only supports the following formats: "
+                for  i in $SUPPORTED_FORMATS  ; do
+                        echo "    "$i
+                done
+                exit 1
+        fi
+        
+        out_path=$(pwd)"/"$path_to_input"/${the_output_format}/"
+        mkdir -p $out_path
+        if [ ! -d "$out_path" ]; then
+                echo >&2 "ERROR: cannot create director at: "$out_path
```

Comment:
--resource /what/ever doesn't work for us, as paths specified in the document always take precedence.
As a workaround, I create softlinks to Openindiana.png in do_workaround, and finally remove these in clean_up; which is working.

Reply:
Just noticed an error while testing. markdown.md produces a LaTeX error for the opensolaris style. It can be reproduced---once i commit and push, with the following command:
    makedocs -vf ./docs/contrib/markdown.md  -s opensolaris

Reply:
I'm also not happy with the name makepdf.sh so I've renamed to makedocs, which is only a link to makepdf.sh; so makepdf.sh will continue to work; but it is possible better to phase out makepdf.sh in the future.

Reply:
There are a few files which have problems using the opensolaris style. This was already the case and is not caused by any of these changes. I believe the problem is caused by not standard markdown which doesn't work correctly with the complex Lua Pandoc script. As it works fine for everything else, fixing this formatting isn't a priority.